core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -455,7 +459,18 @@ object ProducerStateManager {
 
   private def deleteSnapshotFiles(dir: File, predicate: Long => Boolean = _ => true) {
     listSnapshotFiles(dir).filter(file => predicate(offsetFromFile(file))).foreach { file =>
-      Files.deleteIfExists(file.toPath)
+      try{
+        logger.info(s"Sleep before deleteIfExists, thread Id: ${Thread.currentThread().getId}")
+        Thread.sleep(Thread.currentThread().getId % 200)
```

Comment:
Also do we really need to sleep here. Is it not enough to sleep when a exception happens?